main.go

```diff
@@ -33,6 +33,7 @@ func main() {
 	var timeConvert item.Item
 	// set initial time as int64
 	timeConvert.UnixPostTime = 1494505756
+	timeConvert.HoursSincePosting = timeConvert.AddHoursSincePosting()
```

Comment:
Here you're setting up `HoursSincePosting` twice: in this code and in `AddHoursSincePosting`. One of them has to go.
```suggestion
	timeConvert.HoursSincePosting = timeConvert.AddHoursSincePosting()
```

Reply:
Really? I thought I am setting the `HoursSincePosting` based on the UnixPostTime and the time of when the code is run.
To ensure I can test `relativeTime` easily but just setting a `HoursSincePosting` directly.

Reply:
sorry I was unclear, you are storing the result of `time.Since(time.Unix(t.UnixPostTime, 0))` into `HoursSincePosting` twice.
Once in this highlighted block of code, once in the function `AddHoursSincePosting`

It's like you are doing: 
```go

type X { a int }

func (x X) Do() {
  x.a = 1
  return x.a
}

func main(){
  x := X{}

  x.a = x.Do()
}
```

Reply:
Is this still the case? Changed the type to `float64` and all tests pass, main() results both test and live code also looks good `relativeTime` wise.

Reply:
Yes it is. Look at the code, you have twice `.HoursSincePosting = ...`

So you're saving the value once in `AddHoursSincePosting` and once in `main`

```go
func (t Item) AddHoursSincePosting() float64 {
	t.HoursSincePosting = float64((time.Duration(time.Since(time.Unix(t.UnixPostTime, 0))).Hours()))
	return t.HoursSincePosting
}

func main (){
  // ...  
  postUnmarshalled.HoursSincePosting = postUnmarshalled.AddHoursSincePosting()

}

Reply:
Better like this?

Reply:
that works!